maven-core/src/main/java/org/apache/maven/plugin/internal/DefaultMavenPluginManager.java

```diff
@@ -229,7 +228,7 @@ private PluginDescriptor extractPluginDescriptor(Artifact pluginArtifact, Plugin
                 File pluginXml = new File(pluginFile, getPluginDescriptorLocation());
 
                 if (pluginXml.isFile()) {
-                    try (InputStream is = new BufferedInputStream(new FileInputStream(pluginXml))) {
+                    try (InputStream is = Files.newInputStream(pluginXml.toPath())) {
```

Comment:
This is not buffered. Does that matter?

Reply:
No, the woodstox parser will always use an internal buffer.  I think it's best to hand it the raw stream rather than trying to make any assumption on what would be beneficial for it.